zerver/views/registration.py

```diff
@@ -1362,6 +1341,163 @@ def create_realm(request: HttpRequest, confirmation_key: str | None = None) -> H
     )
 
 
+def generate_demo_realm_subdomain() -> str:
+    available_subdomain = False
+    demo_subdomain = ""
+    while not available_subdomain:
+        letters = "".join(random.SystemRandom().choice(string.ascii_lowercase) for _ in range(4))
+        digits = "".join(random.SystemRandom().choice(string.digits) for _ in range(4))
+        demo_subdomain = f"demo-{letters}{digits}"
+        try:
+            check_subdomain_available(demo_subdomain)
+            available_subdomain = True
+        except ValidationError:  # nocoverage
+            continue
+    return demo_subdomain
+
+
+@add_google_analytics
+def create_demo_organization(
+    request: HttpRequest, timezone: Annotated[str, timezone_or_empty_validator()] = ""
+) -> HttpResponse:
+    if not settings.OPEN_REALM_CREATION:
+        return TemplateResponse(
+            request,
+            "zerver/portico_error_pages/demo_creation_disabled.html",
+        )
+    if not settings.CORPORATE_ENABLED:
+        return TemplateResponse(
+            request,
+            "zerver/portico_error_pages/demo_creation_disabled.html",
```

Comment:
Not sure if we want to limit demo organizations by `settings.CORPORATE` or if we want to add another server setting for this feature before moving to production. Also, figured we'd want a different error page for when it's not enabled than the one we use for the `create_realm` view function: `"zerver/portico_error_pages/realm_creation_disabled.html"`.

Reply:
I added a final commit to the series of changes that implements the suggestion above.